src/geometry/euclidean/circumcenter.lean

```diff
@@ -660,12 +660,101 @@ end affine
 
 namespace euclidean_geometry
 
-open affine affine_subspace
+open affine affine_subspace finite_dimensional
 
 variables {V : Type*} {P : Type*} [inner_product_space V] [metric_space P]
     [normed_add_torsor V P]
 include V
 
+/-- Given a nonempty affine subspace, whose direction is complete,
+that contains a set of points, those points are cospherical if and
+only if they are equidistant from some point in that subspace. -/
+lemma cospherical_iff_exists_mem_of_complete {s : affine_subspace ℝ P} {ps : set P} (h : ps ⊆ s)
+  (hn : (s : set P).nonempty) (hc : is_complete (s.direction : set V)) :
+  cospherical ps ↔ ∃ (center ∈ s) (radius : ℝ), ∀ p ∈ ps, dist p center = radius :=
+begin
+  split,
+  { rintro ⟨c, hcr⟩,
+    rw exists_dist_eq_iff_exists_dist_orthogonal_projection_eq h c at hcr,
+    exact ⟨orthogonal_projection s c, orthogonal_projection_mem hn hc _, hcr⟩ },
+  { exact λ ⟨c, hc, hd⟩, ⟨c, hd⟩ }
+end
+
+/-- Given a nonempty affine subspace, whose direction is
+finite-dimensional, that contains a set of points, those points are
+cospherical if and only if they are equidistant from some point in
+that subspace. -/
+lemma cospherical_iff_exists_mem_of_finite_dimensional {s : affine_subspace ℝ P} {ps : set P}
+  (h : ps ⊆ s) (hn : (s : set P).nonempty) [finite_dimensional ℝ s.direction] :
+  cospherical ps ↔ ∃ (center ∈ s) (radius : ℝ), ∀ p ∈ ps, dist p center = radius :=
+cospherical_iff_exists_mem_of_complete h hn (submodule.complete_of_finite_dimensional _)
+
+/-- All n-simplices among cospherical points in an n-dimensional
+subspace have the same circumradius. -/
+lemma exists_circumradius_eq_of_cospherical_subset {s : affine_subspace ℝ P} {ps : set P}
+  (h : ps ⊆ s) (hn : (s : set P).nonempty) {n : ℕ} [finite_dimensional ℝ s.direction]
+  (hd : findim ℝ s.direction = n) (hc : cospherical ps) :
+  ∃ r : ℝ, ∀ sx : simplex ℝ P n, set.range sx.points ⊆ ps → sx.circumradius = r :=
```

Comment:
Wouldn't it be easier for future applications to (also) have versions that take `sx₁` and `sx₂`, and then conclude that they have the same radius?
Idem for the two lemmas below.

Reply:
I've added those versions.